tree 1f86289558ba
parent 32fc6116db71
author wolfmanfx 1343045396 -7200
committer wolfmanfx 1343045396 -7200
revision 3688
branch v1-8

Fix [541]: In Rev 3513 the warning texture is created with a invalid context so
the gl id was 0 - this broke materials without valid textures (if no texture is
bounded the warning texture is bound)
diff --git a/RenderSystems/GL/include/OgreGLTextureManager.h b/RenderSystems/GL/include/OgreGLTextureManager.h
--- a/RenderSystems/GL/include/OgreGLTextureManager.h
+++ b/RenderSystems/GL/include/OgreGLTextureManager.h
@@ -52,6 +52,8 @@
             bool preciseFormatOnly = false);
 
     protected:
+		friend class GLRenderSystem;
+
         /// @copydoc ResourceManager::createImpl
         Resource* createImpl(const String& name, ResourceHandle handle, 
             const String& group, bool isManual, ManualResourceLoader* loader, 
diff --git a/RenderSystems/GL/src/OgreGLRenderSystem.cpp b/RenderSystems/GL/src/OgreGLRenderSystem.cpp
--- a/RenderSystems/GL/src/OgreGLRenderSystem.cpp
+++ b/RenderSystems/GL/src/OgreGLRenderSystem.cpp
@@ -3495,6 +3495,8 @@
 				LogManager::getSingleton().logMessage("Using FSAA from GL_ARB_multisample extension.");
 			}            
 		}
+
+		static_cast<GLTextureManager*>(mTextureManager)->createWarningTexture();
 	}
 	//---------------------------------------------------------------------
 	void GLRenderSystem::_switchContext(GLContext *context)
diff --git a/RenderSystems/GL/src/OgreGLTextureManager.cpp b/RenderSystems/GL/src/OgreGLTextureManager.cpp
--- a/RenderSystems/GL/src/OgreGLTextureManager.cpp
+++ b/RenderSystems/GL/src/OgreGLTextureManager.cpp
@@ -38,8 +38,6 @@
     {
         // register with group manager
         ResourceGroupManager::getSingleton()._registerResourceManager(mResourceType, this);
-
-		createWarningTexture();
     }
     //-----------------------------------------------------------------------------
     GLTextureManager::~GLTextureManager()
